BTSPocket/Controllers/Project/ProjectsViewController.swift

```diff
@@ -8,21 +8,166 @@
+                } else {
+                    self?.tableViewProjects.displayBackgroundMessage(message: "No projects found".localized)
+                    MessageManager.shared.showBar(title: "Info", subtitle: "Cannot get projects".localized, type: .info, containsIcon: true, fromBottom: false)
+                }
+            }
+        }
+    }
+}
 
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+extension ProjectsViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.projectsByUser?.count ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withClass: ProjectTableViewCell.self)
+        let project = self.projectsByUser?[indexPath.row]
+        cell.setProjectValues(project)
+        return cell
     }
-    */
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let lastElement = (self.projectsByUser?.count ?? 1) - 2
+        let nextPage = self.currentPage + 1
+        
+        if indexPath.row == lastElement,
+           nextPage <= self.totalPages {
+            self.getProjectsByUser(.nextPage(nextPage, searchController.searchBar.text?.trim() ?? ""))
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        self.view.endEditing(true)
+        tableView.deselectRow(at: indexPath, animated: true)
+        
+        let vcTeam = Storyboard.getInstanceOf(TeamViewController.self)
+        vcTeam.mode = .projectUsers
+        vcTeam.project = projectsByUser?[indexPath.row]
+        vcTeam.navigationItem.searchController = nil
+        self.navigationController?.pushViewController(vcTeam, animated: true)
+    }
+}
 
+extension ProjectsViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if let text = searchBar.text {
+            if !text.isEmpty {
+                if text.count >= 3 {
+                    NSObject.cancelPreviousPerformRequests(withTarget: self, selector: #selector(self.reload), object: searchBar)
+                    perform(#selector(self.reload), with: searchBar, afterDelay: 1.0)
+                }
+            } else {
+                self.getProjectsByUser(.firstPage)
+            }
+        }
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        if let text = searchBar.text?.trim() {
+            if !text.isEmpty {
+                if text.count < 3 {
```

Comment:
Make value 3 into a constant for easy management. 
Sample: 

`let kMinimumCharactersForSearch: Int = 3`